### ISSUE TITLE
why use sigmoid cross entropy as the loss function in Discriminator?

### ISSUE THREAD

**Issue by LinBornRain** [andrewliao11/gail-tf#5]
It's interesting to see the loss function trick in training the discriminator:
    z * -log(sigmoid(x)) + (1 - z) * -log(1 - sigmoid(x))
May I ask why u use the sigmoid instead ?
And in this code, u use (s,a) as sample to train the discriminator , it can be a high probability that existing a deference between the s generated and s expert, I guess, which may lead to a influence to the confidence of the discriminator. I wonder whether its better to use the (a|s) actions from same state to train the discriminator?

**Comment by andrewliao11** [comment]
from the algorithm1 in [paper](https://arxiv.org/pdf/1606.03476.pdf), it said, "using the TRPO rule with cost function log(D_{wi+1} (s, a))." So using log sigmoid isn't a trick, I just follow the paper :stuck_out_tongue:

For the second problem, you mean feeding single state `(s, a)` might result in confusion of discriminator, right? If so, I agree. While using the whole trajectories might be helpful, it suffers high variances. (just like the relation btn TD and MC) 

I don't think putting (a|s) is a good idea. In this case, `s` is given so that the discriminator can't differentiate T/F from state distribution. though in GAIL paper, the gradient of `s` actually doesn't contribute to the policy network. (See 3.1 in this [paper](http://proceedings.mlr.press/v70/baram17a/baram17a.pdf))

**Comment by LinBornRain** [comment]
Thanks for your links!
Having read GAIL one more time, I totally understand the meaning about logits and sigmoid. Also, the second paper is really inspired! It helps me understanding the variance generated in gradient estimation. And the work in MAIL kind of reminds me about the Value Iteration Net, guess there are more effort should be down in next days in model-based domain ! 
another thing: there is a tiny typo in ur README. It happens in the step EVALUATE YOUR BC AGENT '--task evaluate'
Thanks Anyway!!!

**Comment by andrewliao11** [comment]
you're welcomed
